flink-scala/pom.xml

```diff
@@ -236,4 +230,23 @@ under the License.
 		</plugins>
 	</build>
 
+	<profiles>
+		<profile>
+			<id>scala-2.10</id>
+			<activation>
+				<property>
+					<!-- this is the default scala profile -->
+					<name>!scala-2.11.profile</name>
```

Comment:
I'd prefere -Dscala-2.11 without the profile appended.

On Wed, Mar 11, 2015 at 2:19 PM, Alexander Alexandrov <
notifications@github.com> wrote:

> In flink-scala/pom.xml
> <https://github.com/apache/flink/pull/477#discussion_r26210554>:
>
> > @@ -236,4 +230,23 @@ under the License.
> >  		</plugins>
> >  	</build>
> >
> > +	<profiles>
> > +		<profile>
> > +			<id>scala-2.10</id>
> > +			<activation>
> > +				<property>
> > +					<!-- this is the default scala profile -->
> > +					<name>!scala-2.11.profile</name>
>
> I have created two profiles scala-2.10 and scala-2.11 and configured the
> activation to be mutually exclusive based a dedicated environment variable (
> scala-2.11.profile, but could be changed to scala-2.11). If you want to
> build with with 2.10 or 2.11, you do:
>
> mvn package # for 2.10
> mvn package -Dscala-2.11.profile # for 2.11
>
> The 2.10 profile is (implicitly) activated by default at the moment
> because the activation environment variable scala-2.11.profile is not set
> by default.
>
> I can rewrite it as you suggested (explicit activation based on profile
> names), but then the syntax for building with 2.11 becomes somewhat more
> cumbersome:
>
> mvn package # for 2.10
> mvn package -P!scala-2.10,scala-2.11 # for 2.11
>
> Bare in mind, the current setup does not prohibit you to forcefully
> activate / deactivate profiles from the IDE or based on their names. The
> second set of commands should still work (I will verify this in a minute).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/flink/pull/477/files#r26210554>.
>